src/foam/graphics/TreeGraph.js

```diff
@@ -263,33 +267,72 @@
              slot.set(newValue);
              return false;
            }
-           slot.set((14*slot.get() + newValue)/15);
+           slot.set((2*slot.get() + newValue)/3);
            return true;
          }
        ],
 
        listeners: [
-         {
-           name: 'doLayout',
-           isFramed: true,
-           documentation: 'Animate layout until positions stabilize',
-           code: function() {
-             var needsLayout = false;
-             // Scale and translate the view to fit in the available window
-             var gw = this.graph.width-110;
-             var w  = this.maxRight - this.maxLeft + 55;
-             if ( w > gw ) {
-               var scaleX = Math.min(1, gw / w);
-               needsLayout = this.convergeTo(this.scaleX$, scaleX) || needsLayout;
-             }
-
-             var x = (-this.maxLeft+25)/w * gw + 55;
-             needsLayout = this.convergeTo(this.x$, x) || needsLayout;
-             if ( this.layout() || needsLayout ) this.doLayout();
-             this.graph.invalidate();
-           }
-         }
+        {
+          name: 'doLayout',
+          isFramed: true,
+          documentation: 'Animate layout until positions stabilize',
+          code: function() {
+            var needsLayout = false;
+            // Scale and translate the view to fit in the available window
+            var gw = this.graph.width-110;
+            var w  = this.maxRight - this.maxLeft + 55;
+
+            var x = (-this.maxLeft+25)/w * gw + 55;
+            needsLayout = this.convergeTo(this.x$, x) || needsLayout;
+            
+            if ( this.layout() || needsLayout ) {
+              this.doLayout();
+            }
+            else {
+              this.graph.updateCWidth();
+            }
+          
+            this.graph.invalidate();
+          }
+        }
        ]
      }
+   ],
+
+   listeners: [
+     {
+       name: 'updateCWidth',
+       isFramed: true,
+       code: function() {
+        const maxes = {
+          maxLeft: 0,
+          maxRight: 0
+        };
+          
+        function traverseAndCompare(root) {
+          if ( root.maxLeft < maxes.maxLeft ) maxes.maxLeft = root.maxLeft;
+          if ( root.maxRight > maxes.maxRight ) maxes.maxRight = root.maxRight;
+
+          for ( var i = 0; i < root.children.length; i++ ){
+            traverseAndCompare(root.children[i], root.x);
```

Comment:
Doesn't look like you're using the second arg here anymore.

Reply:
True. fixed it.